### PR TITLE
e2e: add Get tests

### PR DESCRIPTION
**Pull request by gyuho**

### CODE REVIEW

File: e2e/etcdctlv3_test.go [hunk 1]
```diff
@@ -186,6 +219,50 @@ func watchTest(cx ctlCtx) {
...
+
+	var (
+		keys = []string{"key1", "key2", "key3"}
+		vals = []string{"val1", "val2", "val3"}
+	)
+
+	for i := range keys {
+		if err := ctlV3Put(cx, keys[i], vals[i]); err != nil {
+			if cx.dialTimeout > 0 && isGRPCTimedout(err) {
+				cx.errc <- fmt.Errorf("put error (%v)", err)
```
Comment (heyitsanthony) [inline e2e/etcdctlv3_test.go]: `cx.t.Fatal(err)`?

File: e2e/etcdctlv3_test.go [hunk 2]
```diff
@@ -186,6 +219,50 @@ func watchTest(cx ctlCtx) {
...
+				return
+			}
+		}
+	}
+
+	if cx.quorum { // wait a bit after PUT requests
+		time.Sleep(500 * time.Millisecond)
```
Comment (heyitsanthony) [inline e2e/etcdctlv3_test.go]: shouldn't need to sleep if issuing quorum gets

File: e2e/etcdctlv3_test.go [hunk 3]
```diff
@@ -186,6 +219,50 @@ func watchTest(cx ctlCtx) {
...
+		pairs = []string{"key1", "val1"}
+	}
+	if cx.limit > 0 && cx.limit < len(keys) && cx.limit*2 < len(pairs) {
+		pairs = pairs[:cx.limit*2]
+	}
+	// TODO: configure order, sort-by
+
+	if err := ctlV3Get(cx, keyToGet, pairs...); err != nil {
+		if cx.dialTimeout > 0 && isGRPCTimedout(err) {
+			cx.errc <- fmt.Errorf("get error (%v)", err)
```
Comment (heyitsanthony) [inline e2e/etcdctlv3_test.go]: `cx.t.Fatal(err)`?

File: e2e/etcdctlv3_test.go [hunk 4]
```diff
@@ -211,12 +288,20 @@ func ctlV3Put(cx ctlCtx, key, value string) error {
 	return spawnWithExpect(cmdArgs, "OK")
 }
 
-func ctlV3Get(cx ctlCtx, key, value string) error {
+func ctlV3Get(cx ctlCtx, key string, pairs ...string) error {
```
Comment (heyitsanthony) [inline e2e/etcdctlv3_test.go]: pairs should probably be an array of a data type like `type kv struct { key string; val string; }` instead of having a string array; a flat array of strings is sort of confusing for something called "pairs"

Comment (gyuho) [inline e2e/etcdctlv3_test.go]: Agree. `struct` seems like the better idea. Will fix. Thanks!

File: e2e/etcdctlv3_test.go [hunk 5]
```diff
@@ -211,12 +279,20 @@ func ctlV3Put(cx ctlCtx, key, value string) error {
...
 	cmdArgs := append(ctlV3PrefixArgs(cx.epc, cx.dialTimeout), "get", key)
-	if !cx.quorum {
+	if cx.serializable {
```
Comment (heyitsanthony) [inline e2e/etcdctlv3_test.go]: why the change from `quorum` to `serializable`?

**Comment by gyuho** [comment]
@heyitsanthony Fixed up.

1. removed `serializable` option (because we have `quorum`  to configure this)
2. updates error handling
3. changes to `kv struct` for `getTest`

PTAL.

Thanks!

File: e2e/etcdctlv3_test.go [hunk 6]
```diff
@@ -32,6 +32,20 @@ func TestCtlV3PutAutoTLS(t *testing.T)       { testCtl(t, putTest, withCfg(confi
 func TestCtlV3PutPeerTLS(t *testing.T)       { testCtl(t, putTest, withCfg(configPeerTLS)) }
 func TestCtlV3PutClientTLS(t *testing.T)     { testCtl(t, putTest, withCfg(configClientTLS)) }
 
+func TestCtlV3Get(t *testing.T)            { testCtl(t, getTest) }
+func TestCtlV3GetPrefix(t *testing.T)      { testCtl(t, getTest, withPrefix()) }
+func TestCtlV3GetPrefixLimit(t *testing.T) { testCtl(t, getTest, withPrefix(), withLimit(2)) }
+func TestCtlV3GetQuorum(t *testing.T)      { testCtl(t, getTest, withQuorum()) }
+func TestCtlV3GetQuorumAutoTLS(t *testing.T) {
```
Comment (heyitsanthony) [inline e2e/etcdctlv3_test.go]: should every command test tls? it costs about 2 seconds for each e2e test...

Comment (gyuho) [inline e2e/etcdctlv3_test.go]: Sure, I will set TLS by default.

**Comment by gyuho** [comment]
@heyitsanthony Just kept only Auto TLS cases. PTAL. Thanks!

File: e2e/etcdctlv3_test.go [hunk 7]
```diff
@@ -25,34 +25,28 @@ import (
 	"github.com/coreos/etcd/pkg/testutil"
 )
 
-func TestCtlV3Put(t *testing.T)              { testCtl(t, putTest) }
-func TestCtlV3PutTimeout(t *testing.T)       { testCtl(t, putTest, withDialTimeout(0)) }
-func TestCtlV3PutTimeoutQuorum(t *testing.T) { testCtl(t, putTest, withDialTimeout(0), withQuorum()) }
-func TestCtlV3PutAutoTLS(t *testing.T)       { testCtl(t, putTest, withCfg(configAutoTLS)) }
```
Comment (heyitsanthony) [inline e2e/etcdctlv3_test.go]: please keep the different tls configurations for this one command so that there's a baseline full tls test

File: e2e/etcdctlv3_test.go [hunk 8]
```diff
@@ -25,34 +25,28 @@ import (
...
-func TestCtlV3PutPeerTLS(t *testing.T)       { testCtl(t, putTest, withCfg(configPeerTLS)) }
-func TestCtlV3PutClientTLS(t *testing.T)     { testCtl(t, putTest, withCfg(configClientTLS)) }
-
-func TestCtlV3Watch(t *testing.T)        { testCtl(t, watchTest) }
-func TestCtlV3WatchAutoTLS(t *testing.T) { testCtl(t, watchTest, withCfg(configAutoTLS)) }
```
Comment (heyitsanthony) [inline e2e/etcdctlv3_test.go]: probably worth keeping the different varieties of TLS for watch too since they use streams

File: e2e/etcdctlv3_test.go [hunk 9]
```diff
@@ -25,34 +25,28 @@ import (
...
-
-// TODO: Watch with client TLS is not working
-// func TestCtlV3WatchClientTLS(t *testing.T) {
-// 	testCtl(t, watchTest, withCfg(configClientTLS))
-// }
-
-func TestCtlV3WatchInteractive(t *testing.T) { testCtl(t, watchTest, withInteractive()) }
-func TestCtlV3WatchInteractiveAutoTLS(t *testing.T) {
-	testCtl(t, watchTest, withInteractive(), withCfg(configAutoTLS))
+func TestCtlV3PutAutoTLS(t *testing.T) { testCtl(t, putTest, withCfg(configAutoTLS)) }
```
Comment (heyitsanthony) [inline e2e/etcdctlv3_test.go]: make AutoTLS the default in `testCtl` instead of putting `AutoTLS` everywhere?

Comment (gyuho) [inline e2e/etcdctlv3_test.go]: Ok will fix.

**Comment by heyitsanthony** [comment]
lgtm following fixups

File: e2e/etcdctlv3_test.go (hunk 7 above)
Comment (gyuho) [inline e2e/etcdctlv3_test.go]: Oh got it. Will keep the baseline ones.

**Comment by gyuho** [comment]
Fixed up. Will merge in green lights. Thanks!
